pkg/channel/consolidated/reconciler/controller/controller.go

```diff
@@ -85,6 +97,16 @@ func NewController(
 
 	impl := kafkaChannelReconciler.NewImpl(ctx, r)
 
+	statusProber := status.NewProber(
+		logger.Named("status-manager"),
+		NewProbeTargetLister(logger, endpointsInformer.Lister()),
+		func(c v1beta1.KafkaChannel, s eventingduckv1.SubscriberSpec) {
+			logger.Debugf("Ready callback triggered for channel: %s/%s subscription: %s", c.Namespace, c.Name, string(s.UID))
+			impl.EnqueueKey(types.NamespacedName{Namespace: c.Namespace, Name: c.Name})
```

Comment:
I might be wrong here, but shouldn't we enqueue the subscription (which is the one where we want to flip the ready bit)?

Reply:
our `ReconcileKind` is for KafkaChannel. So we must enqueue that to which will eventually reconcile the `spec.status.subscribers` that will end up creating a `patch` only for the desired subscription.